File: app/src/main/java/me/gilo/wc/di/FirebaseModule.java
package me.gilo.wc.di;

import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import dagger.Module;
import dagger.Provides;

import javax.inject.Named;
import javax.inject.Singleton;


@Module
class FirebaseModule {

    @Singleton
    @Provides
    FirebaseFirestore providesFirestore() {
        FirebaseFirestore db = FirebaseFirestore.getInstance();
        return db;
    }

    @Singleton
    @Provides
    @Named("storage")
    StorageReference providesStorage() {
        return FirebaseStorage.getInstance().getReference();
    }

    @Singleton
    @Provides
    @Named("users")
    CollectionReference providesUsers() {
        return getFirestoreInstance().collection("users");
    }

    FirebaseFirestore getFirestoreInstance(){
        FirebaseFirestore db = FirebaseFirestore.getInstance();

        return db;
    }



}

File: app/src/main/java/me/gilo/woodroid/app/di/FirebaseModule.kt
package me.gilo.woodroid.app.di

import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference
import dagger.Module
import dagger.Provides

import javax.inject.Named
import javax.inject.Singleton


@Module
internal class FirebaseModule {

    private val firestoreInstance: FirebaseFirestore
        get() = FirebaseFirestore.getInstance()

    @Singleton
    @Provides
    fun providesFirestore(): FirebaseFirestore {
        return FirebaseFirestore.getInstance()
    }

    @Singleton
    @Provides
    @Named("storage")
    fun providesStorage(): StorageReference {
        return FirebaseStorage.getInstance().reference
    }

    @Singleton
    @Provides
    @Named("users")
    fun providesUsers(): CollectionReference {
        return firestoreInstance.collection("users")
    }


}
